encoders/firebase-encoders-json/src/main/java/com/google/firebase/encoders/FieldDescriptor.java

```diff
@@ -64,13 +63,14 @@ public String getName() {
    * @return {@code T} annotation if present, null otherwise.
    */
   @Nullable
+  @SuppressWarnings("unchecked")
   public <T extends Annotation> T getProperty(@NonNull Class<T> type) {
-    return type.cast(properties.get(type));
+    return (T) properties.get(type);
   }
 
   @NonNull
-  public Set<Annotation> getAllAnnotations() {
-    return new HashSet<>(properties.values());
+  public Collection<Annotation> getAllAnnotations() {
+    return properties.values();
```

Comment:
please also change the constructor to `this.properties = Collections.unmodifiableMap(properties)`

Reply:
the only place called the private constructor is the `build` method, and `Collections.unmodifiableMap(properties)` was applied to the `properties` before passing to the constructor.